main/src/main/java/org/mobilitydata/gtfsvalidator/validator/DateTripsValidator.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2021 Jarvus Innovations LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.validator;
+
+import java.time.LocalDate;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.SortedSet;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+import org.mobilitydata.gtfsvalidator.annotation.GtfsValidator;
+import org.mobilitydata.gtfsvalidator.input.CurrentDateTime;
+import org.mobilitydata.gtfsvalidator.notice.NoticeContainer;
+import org.mobilitydata.gtfsvalidator.notice.SeverityLevel;
+import org.mobilitydata.gtfsvalidator.notice.ValidationNotice;
+import org.mobilitydata.gtfsvalidator.table.GtfsCalendarDateTableContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsCalendarTableContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsFeedInfo;
+import org.mobilitydata.gtfsvalidator.table.GtfsFeedInfoTableContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsTripTableContainer;
+import org.mobilitydata.gtfsvalidator.type.GtfsDate;
+import org.mobilitydata.gtfsvalidator.util.CalendarUtil;
+
+/** */
+@GtfsValidator
+public class DateTripsValidator extends FileValidator {
+
+  private final GtfsCalendarDateTableContainer calendarDateTable;
+  private final GtfsCalendarTableContainer calendarTable;
+  private final GtfsFeedInfoTableContainer feedInfoTable;
+  private final GtfsTripTableContainer tripContainer;
+  private final CurrentDateTime currentDateTime;
+
+  @Inject
+  DateTripsValidator(
+      CurrentDateTime currentDateTime,
+      GtfsCalendarDateTableContainer calendarDateTable,
+      GtfsCalendarTableContainer calendarTable,
+      GtfsFeedInfoTableContainer feedInfoTable,
+      GtfsTripTableContainer tripContainer) {
+    this.currentDateTime = currentDateTime;
+    this.feedInfoTable = feedInfoTable;
+    this.calendarTable = calendarTable;
+    this.calendarDateTable = calendarDateTable;
+    this.tripContainer = tripContainer;
+  }
+
+  @Override
+  public void validate(NoticeContainer noticeContainer) {
+    GtfsDate minStartDate = null;
+    GtfsDate maxEndDate = null;
+
+    GtfsDate serviceWindowStartDate = null;
+    GtfsDate serviceWindowEndDate = null;
+
+    LocalDate now = currentDateTime.getNow().toLocalDate();
+    GtfsDate currentDate = GtfsDate.fromLocalDate(now);
+
+    // Attempt to get Start Date and End Date from feedInfo
+    for (GtfsFeedInfo entity : this.feedInfoTable.getEntities()) {
```

Comment:
feedInfoTable has a `getSingleEntity` method that might be more appropriate to use?